### PR TITLE
[PT2E][Quant][BE] Refactor observer code

### PR DESCRIPTION
**Pull request by kimishpatel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #99106
* #99071
* __->__ #99066
* #99065
* #99064
* #99063

Combine per channel and per tensor observer code

Differential Revision: [D44918494](https://our.internmc.facebook.com/intern/diff/D44918494/)

### CODE REVIEW

**Review by jerryzh168 — APPROVED** [review]

File: torch/ao/quantization/_pt2e/quantizer/qnnpack_quantizer.py [hunk 1]
```diff
@@ -66,42 +66,22 @@ def get_supported_symmetric_config_and_operators() -> List[OperatorConfig]:
...
         quant_min=-128,
         quant_max=127,
-        qscheme=torch.per_tensor_affine,
-        is_dynamic=False,
-    )
-    weight_quantization_spec = QuantizationSpec(
-        dtype=torch.int8,
-        quant_min=-127,
-        quant_max=127,
         qscheme=torch.per_tensor_symmetric,
```
Comment (andrewor14) [inline torch/ao/quantization/_pt2e/quantizer/qnnpack_quantizer.py:74]: @kimishpatel @jerryzh168 This PR changed the activation qscheme from `per_tensor_affine` to `per_tensor_symmetric`. Is this intended? I don't think this matches the corresponding qconfig anymore (https://github.com/pytorch/pytorch/blob/6d5040a1ace7583862992bd1a1b606e873557fe1/torch/ao/quantization/qconfig.py#L288, which uses `per_tensor_affine`)

Comment (andrewor14) [inline torch/ao/quantization/_pt2e/quantizer/qnnpack_quantizer.py:74]: @kimishpatel @jerryzh168 This PR changed the activation qscheme from `per_tensor_affine` to `per_tensor_symmetric`. Is this intended? I don't think this matches the corresponding qconfig anymore, which uses `per_tensor_affine`: https://github.com/pytorch/pytorch/blob/6d5040a1ace7583862992bd1a1b606e873557fe1/torch/ao/quantization/qconfig.py#L288

Comment (andrewor14) [reply torch/ao/quantization/_pt2e/quantizer/qnnpack_quantizer.py:74]: Sent a PR: https://github.com/pytorch/pytorch/pull/99845

Comment (kimishpatel) [reply torch/ao/quantization/_pt2e/quantizer/qnnpack_quantizer.py:74]: oh you are right. It should be per_tensor_affine. Not sure though why tests didnt fail. Thats a bit problematic.

**Pull request closed without merging**